DNN Platform/Tests/DotNetNuke.Tests.Core/Security/PortalSecurity/PortalSecurityTest.cs

```diff
@@ -0,0 +1,52 @@
+﻿using NUnit.Framework;
+using DotNetNuke.Security;
+
+
+namespace DotNetNuke.Tests.Core.Security.PortalSecurity
+{
+    [TestFixture]
+    public class PortalSecurityTest
+    {
+        #region Setup
+        [SetUp]
+        public void Setup()
+        {
+            //TODO: implement environemnt set up
+        }
+
+        #endregion
+
+        #region Tear Down
+        [TearDown]
+        public void TearDown()
+        {
+            //TODO:implement clean up
+        }
+        #endregion
+
+        #region Test Methods
+        [Test]
+        public void Typing_Html_Source_Tag_Should_Not_Be_Allowed()
+        {
+            //Arrange
+            var portalSecurity = new DotNetNuke.Security.PortalSecurity();
+            var htmlEntry = @"<source>
+                               Hi this is personal site
+                               Test for source tag  
+                              </source> ";
+            var markup = DotNetNuke.Security.PortalSecurity.FilterFlag.NoMarkup |
+                         DotNetNuke.Security.PortalSecurity.FilterFlag.NoScripting |
+                         DotNetNuke.Security.PortalSecurity.FilterFlag.NoAngleBrackets;
+            var expectedOutput = "  ";
```

Comment:
IMO, we need to have 3 or more separate tests for various combinations settings for these flags:
- none of these flags set; should return the same without filtering
- NoMarkup
- NoScripting
- NoAngleBrackets
- combination of 2 of these together (3 cases)
- all of them together (which is this test).